torch/csrc/jit/frontend/concrete_module_type.h

```diff
@@ -74,14 +74,19 @@ class VISIBILITY_HIDDEN ConcreteModuleTypeBuilder {
       const TypePtr& type,
       py::object pyFunction);
 
-  void addModule(std::string name, std::shared_ptr<ConcreteModuleType> meta);
+  void addModule(
+      std::string name,
+      std::shared_ptr<ConcreteModuleType> meta,
+      TypePtr hint = nullptr);
 
   void addOverload(
       std::string methodName,
       std::vector<std::string> overloadedMethodNames);
   void addBuiltinFunction(std::string name, std::string symbol_name);
   void addFailedAttribute(std::string name, std::string failureReason);
   void setIterableModuleKind(IterableModuleKind kind);
+  // Set a type hint on this Module (perhaps an InterfaceType).
+  void setHint(TypePtr hint);
```

Comment:
Same here, this is a very generic api, i wouldn't know what it is used for if not for the context i'm reading in now. Could we make it more specific ? `setContainedModulesTypeHint` maybe ? i'm not sure